cart/interfaces/graphql/dto/additional_data.go

```diff
@@ -0,0 +1,72 @@
+package dto
+
+type (
+
+	// CustomAttributes represents map of custom attributes of cart and delivery info
+	CustomAttributes struct {
+		Attributes map[string]string
+	}
+
+	// CustomAttribute that represents a custom attribute of cart
+	CustomAttribute interface {
+		Key() string
+		SetKey(key string)
+	}
+
+	// CustomAttributeBoolean that represents a boolean custom attribute of cart
+	CustomAttributeBoolean struct {
+		key   string
+		Value bool
+	}
+
+	// CustomAttributeString that represents a string custom attribute of cart
+	CustomAttributeString struct {
+		key   string
+		Value string
+	}
+)
+
+// Get attribute by key
+func (c *CustomAttributes) Get(key string) CustomAttribute {
+	if c.Attributes == nil {
+		return nil
+	}
+
+	if value, found := c.Attributes[key]; found {
+		var attribute CustomAttribute
+
+		switch value {
```

Comment:
why not using: `strconv.ParseBool(value)` here?

Reply:
because it should not convert the strings 0 or 1 as booleans in case these values are saved as non boolean attributes

Reply:
Mhh okay, in general, I kinda dislike the approach when you put in a string and get a bool back. Especially because the mutation only accepts strings but then can return bool. Do we really need that type casting or should that be done in the consumer?